allennlp/models/constituency_parser.py

```diff
@@ -249,21 +252,152 @@ def decode(self, output_dict: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor
         all_sentences = output_dict["tokens"]
         all_pos_tags = output_dict["pos_tags"] if all(output_dict["pos_tags"]) else None
         num_spans = output_dict["num_spans"].data
-        trees = self.construct_trees(all_predictions, all_spans, num_spans, all_sentences, all_pos_tags)
 
         batch_size = all_predictions.size(0)
         output_dict["spans"] = [all_spans[i, :num_spans[i]] for i in range(batch_size)]
         output_dict["class_probabilities"] = [all_predictions[i, :num_spans[i], :] for i in range(batch_size)]
 
-        output_dict["trees"] = trees
+        top_k_trees = self.construct_topk_trees(num_trees=16,
+                                                predictions=all_predictions,
+                                                all_spans=all_spans,
+                                                num_spans=num_spans,
+                                                sentences=all_sentences,
+                                                pos_tags=all_pos_tags)
+
+        output_dict["trees"] = [trees[0] for trees in top_k_trees]
+        output_dict['top_k_trees'] = top_k_trees
         return output_dict
 
-    def construct_trees(self,
+    def compute_k_best(self,
+                       sentence: List[str],
+                       pos_tags: List[str],
+                       label_probabilities: torch.FloatTensor,
+                       span_to_index: Dict[Tuple[int, int], int],
+                       num_trees: List[int],
+                       distinguish_between_labels: bool = False):
+        """
+        :param sentence: The sentence for which top-k parses are being computed.
```

Comment:
Can you use the same formatting for docstrings as the rest of the class? It's important because otherwise our docs don't render.

Reply:
it should look like 

```
        Parameters	
        ----------	
        spans: ``List[SpanInformation]``, required.	
            A list of spans, where each span is a ``namedtuple`` containing the	
            following attributes:	
 	
        start : ``int``	
            The start index of the span.	
        end : ``int``

```